gdal_nodes.cpp

```diff
@@ -303,51 +296,86 @@ void OGRWriterNode::process()
+    if (fcnt == 0) {
+      append             = false;
+      overwrite_dataset  = false;
+      bool tables_in_dsn = manager.substitute_globals(filepath).find(
+                             "tables=") != std::string::npos;
+      if (tables_in_dsn) {
+        printf("You are creating a new table in PostgreSQL, but also specified "
+               "the 'tables=' option in the connection string. GDAL will throw "
+               "and error, the table name will be %s in the public schema, "
+               "unless you also passed the schemas= option.\n",
+               manager.substitute_globals(layername).c_str());
+      }
+    }
+  } else {
+    append            = false;
+    overwrite_dataset = false;
   }
 
-  // Create GDAL feature attributes
   std::unordered_map<std::string, size_t> attr_id_map;
-  int fcnt = poLayer->GetLayerDefn()->GetFieldCount();
-  for (auto& term : poly_input("attributes").sub_terminals()) {
-    std::string name = term->get_name();
-    //see if we need to map this name to another one
-    auto search = output_attribute_names.find(name);
-    if(search != output_attribute_names.end()) {
-      if(search->second.size()!=0)
-        name = search->second;
+  if (not append) {
+    // overwrite or create, so field count needs to reset
+    fcnt = 0;
+    poLayer = poDS->CreateLayer(manager.substitute_globals(layername).c_str(), &oSRS, wkbType, lco);
+    if (poLayer == nullptr) {
+      printf("Layer creation failed for %s.\n",
+             manager.substitute_globals(layername).c_str());
+      exit(1);
     }
-    if (term->accepts_type(typeid(float))) {
-      create_field(poLayer, name, OFTReal);
-      attr_id_map[term->get_name()] = fcnt++;
-    } else if (term->accepts_type(typeid(int))) {
-      create_field(poLayer, name, OFTInteger64);
-      attr_id_map[term->get_name()] = fcnt++;
-    } else if (term->accepts_type(typeid(std::string))) {
-      create_field(poLayer, name, OFTString);
-      attr_id_map[term->get_name()] = fcnt++;
+    // Create GDAL feature attributes
+    for (auto& term : poly_input("attributes").sub_terminals()) {
+      std::string name = term->get_name();
+      //see if we need to map this name to another one
+      auto search = output_attribute_names.find(name);
+      if(search != output_attribute_names.end()) {
+        if(search->second.size()!=0)
+          name = search->second;
+      }
+      if (term->accepts_type(typeid(float))) {
+        create_field(poLayer, name, OFTReal);
+        attr_id_map[term->get_name()] = fcnt++;
+      } else if (term->accepts_type(typeid(int))) {
+        create_field(poLayer, name, OFTInteger64);
+        attr_id_map[term->get_name()] = fcnt++;
+      } else if (term->accepts_type(typeid(std::string))) {
+        create_field(poLayer, name, OFTString);
+        attr_id_map[term->get_name()] = fcnt++;
+      }
+    }
+  }
+  else {
+    // Fields already exist, so we need to map the poly_input("attributes")
```

Comment:
Is there also the hypothetical case that we are appending to a layer with different attributes? Probably very rare, but could give weird results. Not important for now, but I will make a note in the code

Reply:
If I understand what I wrote correctly, in that case the `attr_id_map` in L378-381 would remain empty, and then when trying to access it in L392-400 things would crash.